### PR TITLE
Add test plugin WitherSpawnTest

### PR DESCRIPTION
**Pull request by ryantheleach**
Spams chat with spawn causes, to enable, change chatSpawns to true.

It's a bit crap at the moment, Feel free to suggest changes.

### CODE REVIEW

**Comment by phit** [comment]
add a command to toggle it on and off?

File: testplugins/src/main/java/org/spongepowered/test/WitherSpawnTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import org.spongepowered.api.plugin.Plugin;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.format.TextColors;
+
+import java.util.stream.Collectors;
+
+@Plugin(id = "witherspawntest", name = "Wither Spawn Test", description = "Log Wither Skele Spawn and generation")
+public class WitherSpawnTest {
+
+    final boolean chatSpawns = false;
```
Comment (gabizou) [inline testplugins/src/main/java/org/spongepowered/test/WitherSpawnTest.java]: Command based toggle maybe?

File: testplugins/src/main/java/org/spongepowered/test/WitherSpawnTest.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Listener
+    public void onSpawn(SpawnEntityEvent event) {
+        if(!chatSpawns) return;
```
Comment (gabizou) [inline testplugins/src/main/java/org/spongepowered/test/WitherSpawnTest.java]: Formatting.... spacing....

File: testplugins/src/main/java/org/spongepowered/test/WitherSpawnTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        final String entities = event.getEntities().stream()
+                .map(entity -> entity.getType().getId())
+                .collect(Collectors.joining(", "));
+        Sponge.getServer().getBroadcastChannel().send(
+                Text.of(TextColors.GREEN, "Spawning: ", entities, " ", TextColors.WHITE, event.getCause().toString())
```
Comment (gabizou) [inline testplugins/src/main/java/org/spongepowered/test/WitherSpawnTest.java]: You shouldn't have to call `event.getCause().toString()`, `event.getCause()` should suffice, no?

File: testplugins/src/main/java/org/spongepowered/test/WitherSpawnTest.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        );
+    }
+
+    @Listener
+    public void onConstruct(ConstructEntityEvent.Post event) {
+        if(!chatSpawns) return;
```
Comment (gabizou) [inline testplugins/src/main/java/org/spongepowered/test/WitherSpawnTest.java]: Formatting here as well.

**Review by gabizou — CHANGES_REQUESTED** [review]

**Comment by ryantheleach** [comment]
@gabizou as for the functionality, is it ok as is? Or is there any large features you would think would help?

This was originally PR'd so I could discuss, as it was originally going to be a part of a fix for a bug I was chasing, but was unable to find due to insufficient cause information in spawned mobs, so I quickly shot out a PR before giving up. I havn't had a chance to clean up yet.

**Comment by ImMorpheus** [comment]
Updated and rebased

**Pull request merged**